I2C_eeprom_cyclic_store.h

```diff
@@ -0,0 +1,239 @@
+            for(uint16_t slot = 0; slot < _totalPages; slot++)
+            {
+                _eeprom->writeBlock(slot*_pageSize, (uint8_t *) "\xff\xff\xff\xff", 4);
+            }
+
+            _isEmpty = true;
+            _currentSlot = 0;
+            _isInitialized = true;
+        }
+
+        /**
+         * Read data from the eeprom into an instance. The data is read from
+	 * the current slot of the eeprom.
+         * 
+         * @param buffer the reference to an object to read data for
+         * @return true if data was read successfully, false otherwise
+         */
+        bool read(T& buffer) const { return read(&buffer); }
+
+        /**
+         * Read data from the eeprom into a memory instance. The data is read
+	 * from the current slot of the eeprom.
+         * @param buffer the instance to read data into
+         * @return true if data was read successfully, false otherwise
+         */
+        bool read(T* buffer) const
+        {
+            if(!_isInitialized)
+                return false;
+
+            if(_isEmpty)
+                return false;
+
+            Serial.println("Reading "+String(sizeof(T))+" bytes from eeprom");
+
+            return _eeprom->readBlock((_currentSlot*_bufferPages*_pageSize)+sizeof(_currentVersion), (uint8_t *) buffer, sizeof(T)) == sizeof(T);
+        }
+
+        /**
+         * Write data of object to the next slot in the eeprom. This updates
+	 * the current slot of this instance.
+         * 
+         * @param buffer the reference to the object to write data for
+         * @return true if data was written successfully, false otherwise
+         */
+        bool write(T& buffer) { return write(&buffer); }
+
+        /**
+         * Write data of object to the next slot in the eeprom. This updates
+	 * the current slot of the instance.
+         *
+         * @param buffer the pointer to the instance to write data for
+         * @return true if data was written successfully, false otherwise
+         */
+        bool write(T* buffer)
+        {
+            if(!_isInitialized)
+            {
+                return false;
+            }
+
+            if(_isEmpty)
+            {
+                _currentSlot = 0;
+                _currentVersion = 0;
+            }
+            else
+            {
+                _currentSlot++;
+                _currentVersion++;
+
+                // Wrap around to start if going past end of alotted region
+                uint16_t maxSlots = _totalPages/_bufferPages;
+                if(_currentSlot > maxSlots)
+                    _currentSlot = 0;
+            }
+
+            auto buffer_length = sizeof(_currentVersion)+sizeof(T);
+            uint8_t tmp[buffer_length];
+
+            memcpy(tmp, &_currentVersion, sizeof(_currentVersion));
+            memcpy(tmp+sizeof(_currentVersion), buffer, sizeof(T));
+
+            Serial.println("Writing "+String(buffer_length)+" bytes to eeprom");
```

Comment:
Debugging statements should not be this explicit as it may corrupt the output of the user of the class.
==> use a **#ifdef debug**  construct ?

Reply:
Oh, I am really sorry for what turned out to be a bit of a sloppy pull request. :-/ Those shouldn't have been left in there, I thought I had scrubbed it clean of the debug stuff.

Reply:
No need to apologize, these things happen and that is where reviews are for.